test/Microsoft.ML.Benchmarks/Harness/ProjectGenerator.cs

```diff
@@ -35,7 +42,7 @@ protected override void GenerateProject(BuildPartition buildPartition, Artifacts
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <OutputPath>bin\{buildPartition.BuildConfiguration}</OutputPath>
-    <TargetFramework>{TargetFrameworkMoniker}</TargetFramework>
+    <TargetFramework>{TargetFrameworkMoniker}</TargetFramework>{rid}
```

Comment:
I think that MsBuild ignores empty `<RuntimeIdentifier></RuntimeIdentifier>` so we can remove the condition and the need for extra variable

```suggestion
    <TargetFramework>{TargetFrameworkMoniker}</TargetFramework>
    <RuntimeIdentifier>{runtimeIdentifier}</RuntimeIdentifier>
```